sdk/identity/identity/src/credentials/onBehalfOfCredential.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, TokenCredential, UserAssertion } from "@azure/core-auth";
+
+import { MsalOnBehalfOf } from "../msal/nodeFlows/msalOnBehalfOf";
+import { credentialLogger } from "../util/logging";
+import { trace } from "../util/tracing";
+import { MsalFlow } from "../msal/flows";
+import { OnBehalfOfCredentialOptions } from "./onBehalfOfCredentialOptions";
+
+const logger = credentialLogger("OnBehalfOfCredential");
+
+/**
+ * Enables authentication to Azure Active Directory using the On Behalf Of flow.:
+ * https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow
+ */
+export class OnBehalfOfCredential implements TokenCredential {
```

Comment:
I **really** like how this credential turned out. Great job! 👍

Reply:
I’m keeping this unresolved so I can use it to begin working on Monday with a good mood ☺️☺️☺️